File: 0152-maximum-product-subarray/0152-maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int n = nums.size();
        int prefix = 1, suffix = 1;
        int max_pro = INT_MIN;
        for(int i=0; i<n; i++) {
            if(prefix == 0) prefix = 1;
            if(suffix == 0) suffix = 1;
            prefix = prefix * nums[i];
            suffix = suffix * nums[n-i-1];
            max_pro = max(max_pro , max(suffix,prefix));
        }
        return max_pro;
    }
};

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
       int max_pro = Integer.MIN_VALUE;
       int pro = 1;
       for(int num : nums) {
          pro *= num;
          max_pro = Math.max(max_pro,pro);
          if(pro == 0) pro = 1;
       }
       pro = 1;
       for(int i =nums.length-1; i>=0; i--) {
            pro *= nums[i];
            max_pro = Math.max(max_pro,pro);
            if(pro == 0) pro = 1;
       }
       return max_pro;
    }
}